packages/doenetml-worker-rust/doenetml-core/src/dependency.rs

```diff
@@ -0,0 +1,379 @@
+use std::{cell::RefCell, collections::HashMap, rc::Rc};
+
+use crate::{
+    component::{AttributeName, ComponentEnum, ComponentNode, ComponentProfile},
+    essential_state::{
+        create_essential_data_for, EssentialDataOrigin, EssentialStateVar, InitialEssentialData,
+    },
+    state::{StateVarName, StateVarReadOnlyView, StateVarValue},
+    ComponentChild, ComponentIdx, ExtendSource, StateVarIdx,
+};
+
+/// A DependencyInstruction is used to make a Dependency based on the input document structure
+#[derive(Debug)]
+pub enum DependencyInstruction {
```

Comment:
`DependencyRequest` seems like a better name.

Reply:
Except that I've been using "request" to mean requested values for state variables, as in `DependencyUpdatesRequested` and `StateVariableUpdateRequest`. (I think I'll change the former to `DependencyUpdateRequest` for consistency.)

At the moment, I'm not able to come up with a better name that wouldn't be confused with `DependencyUpdateRequest`, but I'm open to suggestions!

Reply:
I don't think that is such a bad conflict. You have a `DepGetRequest` and `DepSetRequest`. Seems like a common distinction.

Reply:
I tried changing this to `DependencyRequest` and renaming the other to `DependencyValueUpdateRequest` but then ended up reverting that change as it got too confusing due to the `DependencyValueUpdateRequest` then referring to a `request_idx` rather than an `instruction_idx`. With the `DependencyValueUpdateRequest` overlapping with the "instructions", it was too confusing to call the instructions "requests".

I'm happy to revisit, but it became too hard to disambiguate them for now.